cpp/src/arrow/integration/json_internal.cc

```diff
@@ -314,6 +314,20 @@ class SchemaWriter {
     writer_->Int(type.list_size());
   }
 
+  void WriteTypeMetadata(const Decimal32Type& type) {
```

Comment:
I think we don't need to unpack to specific widths
```suggestion
  void WriteTypeMetadata(const DecimalType& type) {
```